golem/core/common.py

```
@@ -166,6 +167,21 @@ def short_node_id(node_id):
     return f'{node_id[:8]}..{node_id[-8:]}'
 
 
+def value_if_enum(x: Any) -> Any:
+    """
+    Helper for conveniently dealing with parameters that may contain either
+    an Enum member or its value.
+
+    Returns the thing it gets unless the value is an enum member - in that case
+    returns its `.value`.
+    """
+
+    if not isinstance(x, Enum):
+        return x
+
+    return x.value
+
+

```

I don't want to support both enums and strings. If we get invalid string, Enum will just throw exception and thats ok.
Please revert all changes that are related to this.

---

I originally made it support both because in that function I'm dealing with a codec name that has not been validated yet. If I put that name in the video codec field in the task and let it run, it fails before it reaches validations. I decided that it would be better for the string to reach validations and fail there than to duplicate the validation logic.

But now I changed it according to your request. The video codec is now validated in the autofill function. To be honest, it turned out to be easier than I initially expected (I just run `VideoCodec` and let it fail if the codec is not in the enum) so maybe that's for the better.